src/server/services/hli/stadspas.ts

```diff
@@ -64,62 +62,98 @@ export async function fetchStadspas(
   return stadspasResponse;
 }
 
-export const fetchStadspasBudgetTransactionsWithVerify =
-  createTransactionFetchFn<StadspasBudgetTransaction[]>(
-    fetchStadspasBudgetTransactions
-  );
-
-export const fetchStadspasAanbiedingenTransactionsWithVerify =
-  createTransactionFetchFn<StadspasAanbiedingenTransactionResponse>(
-    fetchStadspasAanbiedingenTransactions
-  );
-
-function createTransactionFetchFn<T>(
-  fetchTransactionFn: FetchStadspasTransactionsFn<T>
+export async function decryptAndValidateStadspasTransactionsKey(
+  transactionsKeyEncrypted: string,
+  verifySessionId?: AuthProfileAndToken['profile']['sid']
 ) {
-  async function inner(
-    requestID: requestID,
-    transactionsKeyEncrypted: string,
-    budgetCode?: StadspasBudget['code'],
-    verifySessionId?: AuthProfileAndToken['profile']['sid']
-  ) {
-    let payload: string[];
-    try {
-      payload = decrypt(transactionsKeyEncrypted).split(':');
-    } catch (error) {
-      captureException(error);
-      return apiErrorResult(
-        'Bad request: Failed to decrypt transactions key',
-        null,
-        400
-      );
-    }
+  let payload: string[];
+  try {
+    payload = decrypt(transactionsKeyEncrypted).split(':');
+  } catch (error) {
+    captureException(error);
+    return apiErrorResult(
+      'Bad request: Failed to decrypt transactions key',
+      null,
+      400
+    );
+  }
 
-    let sessionID = '';
-    let administratienummer = '';
-    let passNumber = '';
-
-    if (verifySessionId) {
-      [sessionID, administratienummer, passNumber] = payload;
-      if (sessionID !== verifySessionId) {
-        return apiErrorResult('Not authorized', null, 401);
-      }
-    } else {
-      [administratienummer, passNumber] = payload;
-    }
+  let sessionID: AuthProfileAndToken['profile']['sid'];
+  let administratienummer: StadspasAdministratieNummer;
+  let pasnummer: string;
 
-    if (!administratienummer || !passNumber) {
+  if (verifySessionId) {
+    [sessionID, administratienummer, pasnummer] = payload;
+    if (sessionID !== verifySessionId) {
       return apiErrorResult('Not authorized', null, 401);
     }
+  } else {
+    [administratienummer, pasnummer] = payload;
+  }
+
+  if (!administratienummer || !pasnummer) {
+    return apiErrorResult('Not authorized', null, 401);
+  }
+
+  return apiSuccessResult({
+    administratienummer,
+    pasnummer: parseInt(pasnummer, 10) as StadspasFrontend['passNumber'],
+  });
+}
+
+export async function decryptAndFetch<T>(
```

Comment:
Heeft alleen referenties in dit bestand
```suggestion
async function decryptAndFetch<T>(
```